### PR TITLE
Added prettyPrintJson flag to HttpLogOptions (#6439)

### PR DESCRIPTION
**Pull request by evgeniycheban**

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java [hunk 1]
```diff
@@ -183,4 +184,25 @@ public HttpLogOptions setApplicationId(final String applicationId) {
         }
         return this;
     }
+
+    /**
+     * Gets flag to allow pretty printing of JSON message bodies.
+     *
+     * @return true if pretty printing of JSON message bodies is allowed.
+     */
+    public boolean isPrettyPrintJson() {
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java]: I know the original issue stated to make the pretty print flag settable, but given that there are other formats such as `XML` could this be called `prettyPrintBody`?

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java [hunk 2]
```diff
@@ -43,26 +43,14 @@
...
      */
-    private HttpLoggingPolicy(HttpLogOptions httpLogOptions, boolean prettyPrintJson) {
-        this.prettyPrintJson = prettyPrintJson;
-
+    public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
         if (httpLogOptions == null) {
             this.httpLogDetailLevel = HttpLogDetailLevel.NONE;
             this.allowedHeaderNames = Collections.emptySet();
             this.allowedQueryParameterNames = Collections.emptySet();
+            this.prettyPrintJson = false;
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java]: Along with the other comment, this property name could be changed to be more generic.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java (hunk 1 above)
Comment (evgeniycheban) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java]: Renamed to `prettyPrintBody`

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java (hunk 2 above)
Comment (evgeniycheban) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java]: Renamed this property as well.

**Comment by alzimmermsft** [comment]
Thank you for submitting this PR @evgeniycheban, I'll go ahead and merge this in for you! 😄

**Pull request merged**